src/main/java/frc/robot/subsystems/ArmControlSubsystem.java

```diff
@@ -0,0 +1,325 @@
+import com.revrobotics.CANSparkMax.IdleMode;
+import com.revrobotics.CANSparkMaxLowLevel.MotorType;
+import edu.wpi.first.math.MathUtil;
+import edu.wpi.first.math.controller.PIDController;
+import edu.wpi.first.math.filter.SlewRateLimiter;
+import edu.wpi.first.math.util.Units;
+import edu.wpi.first.wpilibj.DutyCycleEncoder;
+import edu.wpi.first.wpilibj.smartdashboard.SendableChooser;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+import edu.wpi.first.wpilibj2.command.Command;
+import edu.wpi.first.wpilibj2.command.FunctionalCommand;
+import edu.wpi.first.wpilibj2.command.PrintCommand;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.constants.ArmConstants;
+import frc.robot.constants.CompConstants;
+
+
+public class ArmControlSubsystem extends SubsystemBase {
+
+  public enum ArmMotorMode {
+    COAST, BRAKE, OFF
+  }
+
+  private final WPI_TalonFX mLeftPivotController = new WPI_TalonFX(ArmConstants.LEFT_PIV_MOTOR_ID);
+  private final WPI_TalonFX mRightPivotController =
+      new WPI_TalonFX(ArmConstants.RIGHT_PIV_MOTOR_ID);
+  private final CANSparkMax mExtensionController =
+      new CANSparkMax(ArmConstants.EXT_SPARK_ID, MotorType.kBrushless);
+
+  private final DutyCycleEncoder mAbsPivEncoder = new DutyCycleEncoder(ArmConstants.DIOPORTPIV);
+  private final RelativeEncoder mExtensionEncoder = mExtensionController.getEncoder();
+
+  private final SlewRateLimiter mPivotRateLimiter;
+  private final PIDController mExtensionPID;
+  private final PIDController mPivotPID;
+
+  // This is called Ultra Instinct because setting this boolean to true removes the clamps
+  // so that if the belt skips, the operator can still run manually without the bad offsets
+  private boolean mUltraInstinct = false;
+
+  private double mPivotRelEncoderOffsetRot;
+  private double mCurrentPivotRotation;
+  private double mDesiredPivotRotation = ArmConstants.MIN_PIV_ANGLE_RAD;
+
+  private double mCurrentExtensionDistance = ArmConstants.ZERO_EXTENSION_IN;
+  private double mDesiredExtensionDistance = ArmConstants.MIN_EXT_LEN_IN;
+
+
+
+  private boolean mIsInitialized = false;
+
+  private final SendableChooser<ArmMotorMode> mChooser = new SendableChooser<>();
+
+
+  public ArmControlSubsystem() {
+
+    mPivotPID = new PIDController(1.4, 0, 0);
+    mPivotPID.setTolerance(ArmConstants.RESTING_PIV_TOLERANCE_RAD);
+
+    mExtensionPID = new PIDController(0.19, 0, 0);
+    mExtensionPID.setTolerance(ArmConstants.RESTING_EXT_TOLERANCE_IN);
+
+    mPivotRateLimiter = new SlewRateLimiter(ArmConstants.MAX_PIV_RATE_RAD_SEC);
+
+    mAbsPivEncoder.setConnectedFrequencyThreshold(ArmConstants.CONNECTION_THRESH_HZ);
+
+    mPivotRelEncoderOffsetRot =
+        -mAbsPivEncoder.getAbsolutePosition() * ArmConstants.PIVOT_ABS_ENC_TO_ROTATION
+            + ArmConstants.PIV_INIT_OFFSET_ROT;
+
+    setDefaultConfig();
+
+    SmartDashboard.putNumber("kG", ArmConstants.KG);
+
+    mChooser.addOption("Brake", ArmMotorMode.BRAKE);
```

Comment:
Redesign the arm sybsystem so that it works and only expose high level commands to robot operators. Operators are busy keeping the robot operating at fast pace. Dont ask them think at brake mode vs coast mode level for either pivot or extention motors.

Reply:
During the competition, we need to switch between brake and coast mode and many of those smaller commands multiple times. For example, switching to brake/coast mode is necessary while loading into the field to line up the arm, lower the arm after the round, and just generally while debugging issues. During competition, while those options are there, the operator doesn't need to deal with those commands, since none are mapped to the controller.